src/schema/posts.ts

```diff
@@ -1582,6 +1635,91 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
     },
   },
   Mutation: {
+    createSourcePostModeration: async (
+      _,
+      {
+        sourceId,
+        image,
+        title,
+        content,
+        commentary,
+        type,
+        sharedPostId,
+        imageUrl,
+        externalLink,
+      }: CreateSourcePostModerationArgs,
+      ctx: AuthContext,
+      info,
+    ): Promise<GQLSourcePostModeration> => {
+      const { con, userId } = ctx;
+
+      const sourceMember = await con.getRepository(SourceMember).findOne({
+        where: { sourceId: sourceId, userId: userId },
+        relations: ['source'],
+      });
+
+      const source = await sourceMember?.source;
+
+      if (!source) {
+        throw new ForbiddenError('Access denied!');
+      }
+
+      await canAccessSource(
+        ctx,
+        source,
+        sourceMember,
+        SourcePermissions.Post,
+        sourceId,
+      );
+      const id = generateUUID();
```

Comment:
I think `id` should be autogenerated in insert, do we need to do it here?

Reply:
This is how we do it in the other requests where we return the entity that was just created with graphql as far as I can tell

Reply:
Leaving this up to @capJavert we do use both ways.
Think uuid only for article related stuff, but it's more for yggdrasil I think.

Reply:
We don't have to generate UUID, psql will automatically generate it since its a primary column.